plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableName.java

```diff
@@ -81,4 +83,53 @@ public static IcebergTableName from(String name)
 
         return new IcebergTableName(table, type);
     }
+
+    public static String tableNameFrom(String name)
+    {
+        Matcher match = TABLE_PATTERN.matcher(name);
+        if (!match.matches()) {
+            throw new TrinoException(NOT_SUPPORTED, "Invalid Iceberg table name: " + name);
+        }
+
+        return match.group("table");
+    }
+
+    public static Optional<TableType> tableTypeFrom(String name)
+    {
+        Matcher match = TABLE_PATTERN.matcher(name);
+        if (!match.matches()) {
+            throw new TrinoException(NOT_SUPPORTED, "Invalid Iceberg Lake table name: " + name);
+        }
+        String typeString = match.group("type");
+        if (typeString == null) {
+            return Optional.of(DATA);
+        }
+        try {
+            return Optional.of(TableType.valueOf(typeString.toUpperCase(Locale.ENGLISH)));
```

Comment:
nit: Static import `ENGLISH`. Also, can we change existing `Locale.ROOT` to `ENGLISH`?